File: CallyAIbackend/src/main/java/com/example/callyaibackend/dto/FoodLogDtos.java
package com.example.callyaibackend.dto;

import java.time.LocalDateTime;
import java.util.List;

public class FoodLogDtos {
    public static class ItemReq {
        public String name;
        public int caloriesPer100g;
        public int grams;
        public int quantity;
        public Double proteinPer100g;
        public Double fatPer100g;
        public Double carbsPer100g;
    }

    public static class CreateReq {
        public String consumedAt; // "yyyy-MM-dd'T'HH:mm:ss"
        public List<ItemReq> items;
    }

    public static class EntryRes {
        public long id;
        public String name;
        public int calories;
        public int caloriesPer100g;
        public int quantity;
        public int grams;
        public int totalCalories;
        public String consumedAt;
        public double protein;
        public double fat;
        public double carbs;
        public double totalProtein;
        public double totalFat;
        public double totalCarbs;
    }
}

File: app/src/main/java/com/example/callyaiandroid/network/dto/FoodLogDtos.kt
package com.example.callyaiandroid.network.dto

import com.google.gson.annotations.SerializedName

data class FoodLogDtos(
    val id: Long,
    val name: String,
    @SerializedName("calories")
    val portionCalories: Int,
    @SerializedName("caloriesPer100g")
    val caloriesPer100g: Int,
    val grams: Int,
    val quantity: Int,
    @SerializedName(value = "totalCalories", alternate = ["total_calories"])
    val totalCalories: Int?,
    @SerializedName(value = "consumedAt", alternate = ["consumed_at"])
    val consumedAt: String,
    val protein: Double,
    val fat: Double,
    val carbs: Double,
    @SerializedName(value = "totalProtein", alternate = ["total_protein"])
    val totalProtein: Double,
    @SerializedName(value = "totalFat", alternate = ["total_fat"])
    val totalFat: Double,
    @SerializedName(value = "totalCarbs", alternate = ["total_carbs"])
    val totalCarbs: Double,
)
